src/app/content/search/hooks/index.ts

```diff
@@ -74,7 +74,7 @@ export const receiveSearchHook: ActionHookBody<typeof receiveSearchResults> = (s
   const page = currentPage || targetPage;
 
   const navigation = {
-    params: getBookPageUrlAndParams(book, targetPage).params,
+    params: (await getBookPageUrlAndParams(book, targetPage, services.bookConfigLoader)).params,
```

Comment:
this is a situation where it would never be valid for it to be async, because the content has to be loaded before we get this far, so the book must be in the config already

something to keep in mind, maybe we can simplify some logic if we build on that assumption

Reply:
@TomWoodward 
I'm thinking about creating a separate functions inside `urlUtils` like: `getBookPageUrlAndParams` and async version `cacheSafeGetBookPageUrlAndParams which uses bookConfigLoader`. But it will cause a lot of code reduplication. 
Generally, it's all about making making the book config loading optional.

Reply:
It would remove changes in `domVisitor` and `urlChecker`